source/js/GliderRecipe.js

```diff
@@ -0,0 +1,201 @@
+  var value;
+  Object.keys(object).some(function (k) {
+    if (k === key) {
+      value = object[k];
+      return true;
+    }
+    if (object[k] && typeof object[k] === 'object') {
+      value = searchForKey(object[k], key);
+      return value !== undefined;
+    }
+  });
+  return value;
+}
+
+/**
+ * Similar to getUrl(), this function extracts the organizations name from the
+ * schema JSON object. It's not in a standard location so this function helps.
+ * @param {Object} data Raw recipe JSON to find the org string of
+ * @returns {String} If found, it retuns the name of the org as a string, otherwise null
+ */
+ function getOrganization(data) {
+  if (data.publisher?.name) return data.publisher?.name;
+  if (data['@graph']) {
+    for (let i = 0; i < data['@graph'].length; i++) {
+      if (data['@graph'][i]['@type'] == 'Organization') {
+        return data['@graph'][i].name;
+      }
+    }
+  };
+  return null;
+}
+
+/**
+ * Similar to getOrganization(), this extracts recipe name from raw JSON
+ * @param {Object} Data Raw recipe JSON to find name of
+ * @returns {String} if found, returns the name of recipe as string, otherwise null
+ */
+function getRecipeTitle(data) {
+  if (data.name) return data.name;
+  let value = null;
+  if (data['@graph']) {
+    data['@graph'].forEach((obj) => {
+      if (obj['@type'] == 'Recipe') {
+        value = obj['name'];
+      }
+    });
+  }
+  return value;
+}
+
+/**
+ * Extract the URL from the given recipe schema JSON object
+ * @param {Object} data Raw recipe JSON to find the URL of
+ * @returns {String} If found, it returns the URL as a string, otherwise null
+ */
+function getUrl(data) {
+  if (data.url) return data.url;
+  if (data['@graph']) {
+    for (let i = 0; i < data['@graph'].length; i++) {
+      if (data['@graph'][i]['@type'] == 'Article') return data['@graph'][i]['@id'];
+    }
+  };
+  return null;
+}
+
+/**
+ * Converts ISO 8061 time strings to regular english time strings.
+ * Not perfect but it works for this lab
+ * @param {String} time time string to format
+ * @return {String} formatted time string
+ */
+function convertTime(time) {
+  let timeStr = '';
+
+  // Remove the 'PT'
+  time = time.slice(2);
+
+  let timeArr = time.split('');
+  if (time.includes('H')) {
+    for (let i = 0; i < timeArr.length; i++) {
+      if (timeArr[i] == 'H') return `${timeStr} hr`;
+      timeStr += timeArr[i];
+    }
+  } else {
+    for (let i = 0; i < timeArr.length; i++) {
+      if (timeArr[i] == 'M') return `${timeStr} min`;
+      timeStr += timeArr[i];
+    }
+  }
+
+  return '';
+}
+
+/**
+ * Takes in a list of ingredients raw from imported data and returns a neatly
+ * formatted comma separated list.
+ * @param {Array} ingredientArr The raw unprocessed array of ingredients from the
+ *                              imported data
+ * @return {String} the string comma separate list of ingredients from the array
+ */
+function createIngredientList(ingredientArr) {
```

Comment:
Codacy has a fix for the issue: [Missing space before function parentheses.](https://app.codacy.com/gh/cse110-fa21-group27/cse110-fa21-group27/pullRequest?prid=8447291)
```suggestion
function createIngredientList (ingredientArr) {
```